src/test/java/fi/helsinki/cs/tmc/core/commands/DownloadAdaptiveExerciseTest.java

```diff
@@ -121,13 +135,15 @@ private void setUpMocks() throws Exception {
     @Test
     public void testDownloadAndExtractSuccessWithRealZip() throws Exception {
         verifyZeroInteractions(langs);
-        TmcServerCommunicationTaskFactory realFactory = new TmcServerCommunicationTaskFactory();
+        TmcServerCommunicationTaskFactory realFactory = new TmcServerCommunicationTaskFactory(settings, oauth);
         assertNotNull(TmcSettingsHolder.get());
         command = new DownloadAdaptiveExercise(mockObserver, realFactory);
+        Path testPath = Paths.get(System.getProperty("user.dir"));
 
-        when(settings.getTmcProjectDirectory()).thenReturn(arithFuncsTempDir);
+        when(settings.getTmcProjectDirectory()).thenReturn(testPath);
 
         Exercise exercise = command.call();
-        assertTrue(Files.exists(arithFuncsTempDir));
+        assertTrue(Files.exists(testPath.resolve(exercise.getCourseName())));
+        FileUtils.deleteDirectory(testPath.resolve(exercise.getCourseName()).toFile());
```

Comment:
Please do not delete stuff from my user directory. Use the `TemporaryFolder` you have set up as a `@Rule`.